wordpress-mobile/WordPress-Android
# Modules Restructuring - Restructure Annotations + Processors Modules

Parent #16562
Fixes #16568
Fixes #16569

This PR restructures the `annotations` and `processors` module by removing the `WordPress` keyword from it and removing any unnecessary files lying around (like `.gitignore`)

The `WordPress` keyword is dropped from theses modules for the below main reasons:
- This repo is already in the context of WordPress, as such, having the `WordPress` prefix added to every module is unnecessary.
- This repo, in addition to the WordPress app, is also building the Jetpack app. As such, remove the word `WordPress` from every module will help avoiding any Jetpack related naming confusion.

PS: @planarvoid I added you as the main reviewer, that is, in addition to the @wordpress-mobile/owl-team team itself, since you where the author of this feature flag mechanism for WPAndroid and will know if something is missing. As such, I think you might be the best person to sign-off on that change for WPAndroid. 🥇 

-----

Question (❓): As per my licensing question on this other related PR of mine [here](https://github.com/wordpress-mobile/WordPress-Android/pull/16744) and since I don't see any `LICENCE` file for this module, like I did for the `analytics` mode, which is pointing to `MIT`, I am not sure if we should include a licence for this module too or exclude the licensing information from the `analytics` and all other modules in general so that we start depending solely on the root level `LICENCE` that WPAndroid is using, which is `GPLv2`, wdyt? 🤔 

-----

To test:

- Build the app and install it on your device or emulator.
- Go to `Stats` and verify that the current non-revamp stats screen is shown.
- Go to `App Settings` -> `Debug Settings`, enable the `StatsRevampV2FeatureConfig` feature flags and click on the `RESTART THE APP` 🟦 button at the bottom.
- Go to `Stats` again and verify that the new revamp stats screen is now shown.
- Go to `App Settings` -> `Debug Settings`, disable the `StatsRevampV2FeatureConfig` feature flags and click on the `RESTART THE APP` 🟦 button at the bottom.
- Go to `Stats` again and verify that the current non-revamp stats screen is shown.
- The above verifies that the annotations/processors are working as expected since the feature flag mechanism is being driven by that.

-----

## Regression Notes
1. Potential unintended areas of impact

N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

PR submission checklist:
